applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.cpp

```diff
@@ -0,0 +1,288 @@
+    Model& rModel,
+    Parameters Parameters)
+{
+    KRATOS_TRY
+
+    Parameters.ValidateAndAssignDefaults(GetDefaultParameters());
+    mpModelPart = &rModel.GetModelPart(Parameters["model_part_name"].GetString());
+    
+    // Reading flow direction
+    const auto flow_direction_name = Parameters["flow_direction_variable"].GetString();
+
+    KRATOS_ERROR_IF_NOT(KratosComponents<VectorVariable>::Has(flow_direction_name))
+        << "specified flow direction variable is not a known vector variable." << std::endl;
+    
+    mpFlowDirectionVariable = & KratosComponents<VectorVariable>::Get(flow_direction_name);
+
+    mRefreshNormalsEveryTimeStep = Parameters["refresh_normals_every_time_step"].GetBool();
+
+    // Reading boundary conditions to enforce
+    mSubsonicBCs.reserve(Parameters["subsonic_boundary_conditions"].size());
+    mSupersonicBCs.reserve(Parameters["supersonic_boundary_conditions"].size());
+
+    for(auto & var_settings : Parameters["subsonic_boundary_conditions"])
+    {
+        ReadBoundaryCondition(mSubsonicBCs, var_settings);
+    }
+
+    for(auto & var_settings : Parameters["supersonic_boundary_conditions"])
+    {
+        ReadBoundaryCondition(mSupersonicBCs, var_settings);
+    }
+
+    KRATOS_CATCH("");
+}
+
+void ApplyMachDependentBoundaryConditions::ExecuteInitialize()
+{
+    NormalCalculationUtils().CalculateUnitNormals<Condition>(*mpModelPart);
+}
+
+
+void ApplyMachDependentBoundaryConditions::ExecuteInitializeSolutionStep()
+{
+    if(mRefreshNormalsEveryTimeStep)
+    {
+        NormalCalculationUtils().CalculateUnitNormals<Condition>(*mpModelPart);
+    }
+
+    // Enabling and disabling BC according to the time and active interval
+    const double time = mpModelPart->GetProcessInfo().GetValue(TIME);
+
+    for(auto & boundary_condition: mSupersonicBCs)
+    {
+        boundary_condition.ActivateIfInsideTimeInterval(time);
+    }
+
+    for(auto & boundary_condition: mSubsonicBCs)
+    {
+        boundary_condition.ActivateIfInsideTimeInterval(time);
+    }
+
+    // Enforcing boundary conditions
+    block_for_each(mpModelPart->Nodes(), [&](NodeType & rNode)
+    {
+        // Computing mach projected onto normal
+        const auto & normal = rNode.FastGetSolutionStepValue(NORMAL);
+        const auto & flow_direction = rNode.FastGetSolutionStepValue(*mpFlowDirectionVariable);
+
+        const double & mach = rNode.GetValue(MACH);
+        const double mach_projection = mach * inner_prod(normal, flow_direction) / norm_2(flow_direction);
+        
+        // Chosing BC set to enforce according to mach
+        const bool supersonic = fabs(mach_projection) >= 1.0;
+        
+        const auto & active_bc  = supersonic ? mSupersonicBCs : mSubsonicBCs;
+        const auto & passive_bc = supersonic ? mSubsonicBCs   : mSupersonicBCs;
+
+        for(const auto & boundary_condition: passive_bc)
+        {
+            boundary_condition.Release(rNode);
+        }
```

Comment:
Just thought that it'll play nicer with other processes if the inactive BC's don't free their Dof here, and instead have all BC's free them during ExecuteFinalize, like similar processes do.

Should be fixed by tomorrow.